books/kestrel/c/syntax/validation-information.lisp

```diff
@@ -700,6 +700,101 @@
                                  ((:e tau-system)))))
   :hooks (:fix))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define type-compatiblep ((x typep) (y typep))
+  :returns (yes/no booleanp)
+  :short "Check that two @(see type)s are compatible [C17:6.2.7]."
+  :long
+  (xdoc::topstring
+   (xdoc::p
+    "Informally, two types are compatible if they represent \"the same\"
+     type.")
+   (xdoc::p
+    "Because we currently only model an approximation of C types, our notion of
+     compatibility is also approximate.
+     Specifically, this relation overapproximates true type compatibility.
+     Compatible types should always be recognized as such,
+     but incompatible types may also be recognized.")
+   (xdoc::p
+    "In particular:"
+    (xdoc::ul
+     (xdoc::li "all composite types of a particular kind are currently
+                considered compatible.")
+     (xdoc::li "type qualifiers are ignored.")
+     (xdoc::li "all types are compatible with the abstract @(':unknown')
+                type.")
+     (xdoc::li "@('enum') types are compatible with "
+               (xdoc::i "all")
+               " integer types (not just one particular type).")))
+   (xdoc::p
+     "Eventually, we shall refine the notion of compatibility alongside our
+      representation of type in order to reflect true type compatibility.
+      This may require an additional argument representing the implementation
+      environment so that we may establish "
+     (xdoc::i "which")
+     " integer type is to be considered compatible with @('enum') types.")
+   (xdoc::p
+    "True type compatibility is an equivalence relation, but our approximate
+     notion of compatibility is not.
+     That is because @('type-compatiblep') is not transitive.
+     For instance, @(':void') is compatible with @(':unknown'), as is
+     @(':bool'), but @(':void') is "
+    (xdoc::i "not")
+    " compatible with @(':bool')."))
+  (b* ((x (type-fix x))
+       (y (type-fix y)))
+    (or (equal x y)
+        (type-case x :unknown)
+        (type-case y :unknown)
+        (and (type-integerp x) (type-case y :enum))
+        (and (type-case x :enum) (type-integerp y))))
+  :hooks (:fix))
+
+;;;;;;;;;;;;;;;;;;;;
```

Comment:
I would use `///` and put the reflexivity and symmetry theorems under the define, so they should up in the XDOC.